packages/ui/src/models/visualization/base-visual-entity.ts

```diff
@@ -155,6 +155,9 @@ export interface IVisualizationNode<T extends IVisualizationNodeData = IVisualiz
 
   /** Retrieve the node's validation status, relying into the underlying entity */
   getNodeValidationText(): string | undefined;
+
+  /** Return extra icons for the CustomGroup (entity-specific) */
+  getGroupIcons?(): { icon: string; title: string }[];
```

Comment:
If we provide a default implementation and update the main interface, we can set this as always provided.

```suggestion
  getGroupIcons(): { icon: string; title: string }[];
```